src/insert_api_to_table.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Sep 13 14:13:34 2021
+
+@author: pkh35
+"""
+import json
+import urllib
+from datetime import datetime
+from urllib.parse import urlparse, parse_qs
+
+import requests
+import sqlalchemy
+import validators
+from bs4 import BeautifulSoup
+
+import setup_environment
+import tables
+
+
+def extract_api_params(api):
+    """to parse the api, to get the base url and params"""
+    parsed_api = urlparse(api)
+    params = parse_qs(parsed_api.query)
+    base_url = urllib.parse.urlunparse(
+        (parsed_api.scheme, parsed_api.netloc, parsed_api.path, "", "", ""))
+    return json.dumps(params), base_url
+
+
+def url_validator(url=None):
+    """check if the url entered by the user is valid"""
+    valid = validators.url(url)
+    return url if valid else print(f"Invalid URL: {url}")
+
+
+def api_data_modified_date(url=None):
+    """to get the modified date of the data source"""
+    try:
+        content = requests.get(url)
+        soup = BeautifulSoup(content.text, 'html.parser')
+        url_update_date = soup.find(
+            "th", text="Last updated").find_next_sibling("td").text
+
+        def mdy_to_ymd(url_date):
+            """to change the format of the date type variable"""
+            return datetime.strptime(url_date, '%d %b %Y').strftime('%Y-%m-%d')
+
+        url_update_date = mdy_to_ymd(url_update_date)
+        return url_update_date
+    except Exception as error:
+        print("An exception has occured:", error, type(error))
```

Comment:
Test what happens when this exception actually occurs. it looks like you ignore it instead of properly handling the exception